share/getter.go

```diff
@@ -96,3 +96,11 @@ func (row *NamespacedRow) verify(rowRoot []byte, namespace Namespace) bool {
 		rowRoot,
 	)
 }
+
+// IsAbsenceProof checks if `NamespacedShares.Proof` proves the absence of the namespace.ID.
+func (ns NamespacedShares) IsAbsenceProof() bool {
+	if len(ns) == 0 || (len(ns[0].Shares) == 0 && ns[0].Proof != nil) {
```

Comment:
So it case of absence proof it gives you one empty share?

Reply:
Feel like leaky abstraction on the GetByNamespace side. If we make it truly nil on the implementation side, then this PR would not be needed and the case would be handled by a general path.

Reply:
> So it case of absence proof it gives you one empty share?

It gives 0 shares and one proof

Reply:
> If we make it truly nil on the implementation side, then this PR would not be needed and the case would be handled by a general path.

I've discussed this with @walldiss. The whole point of the non-nil result is different cases:
* when requested namespace B is not found within the range of namespaces in a single row[A, C] -> in this case we can produce an absence proof.
* When the requested namespace B is not found within multiple rows: Row0[A], Row1[C] -> in this case we can't produce anything and return nil.

Reply:
But I agree that such behavior is unclear and we should do smth about it. But now I'd suggest creating a separate issue and moving the discussion there.

Reply:
For both given cases, we can have a general path. For the user of `GetSharesByNamespace`, there is no difference if it's with absence proof or not, and even if there is, they can derive this information by looking into the DAH header or by looking if the Proof is present on the returned NamespaceShares.

I suggest only never returning an empty share for `GetSharesByNamespace`:
* It's confusing. There is no documentation on what that means, as well as it's not obvious.
* It requires users on top to keep track of this edge case when it can be avoided. (Leaky abstraction)

Reply:
I don't think we should postpone this in the issue either:
* It's not complicated to fix this now
* If we postpone this now, it's very unlikely we will come back to this.

Reply:
Ok, the general path for this is to return nil for both cases. That means if we can't prove the inclusion of the data, then we don't have to prove its absence, correct?

Reply:
For the path when there is an absence of proof, we must return it. We just need to avoid returning non empty shares row

Reply:
With or without proofs, it should return NamespacedShares both of which should succeed when we call Verify on it.

Reply:
ok, so in case if data was not found, the results will be:
* `[]share.NamespacedRow{}`-> when it's not possible to build an Absence Proof;
*` []share.NamespacedRow{Proof: *nmt.Proof}` -> when the Absence Proof exists;

Reply:
But we will anyway need `IsAbsenceProof`

Reply:
We would we need it?

Reply:
I meant not the method itself, but this condition inside the blob module. Because the logic iterates over each row collecting shares and proofs and in this case (`[]share.NamespacedRow{Proof: *nmt.Proof}`), we'll get an error

Reply:
So we actually don't need to do anything else besides the if above.

Reply:
👍 true, but I'd go with `break`

Reply:
break works if we are sure that only one absence proof is possible

Reply:
Get accepts only one namespace, so technically we can't have more then 1 absence proof

Reply:
I don't remember if you can have multiple proofs over multiple rows with one namespace

Reply:
if NamespaceB falls in between Rows -> Row0[NamespaceA], Row1[NamespaceC], where 
`NamespaceA < NamespaceB < NamespaceC`, then no Absence proof is expected.

Reply:
When I was trying to make an example below, I realized that the case below is not possible. Because that would imply B is actually there
* Row1[NamespaceA], Row1[NamespaceB]
* Row2[NamespaceB], Row2[NamespaceC]

Reply:
Break is fine